File: Api/src/main/java/com/alphaz/api/controller/UserController.java
package com.alphaz.api.controller;


import com.alphaz.application.authorization.dto.user.UserModel;
import com.alphaz.application.authorization.dto.user.UserUpdateModel;
import com.alphaz.application.authorization.service.PrivilegeService;
import com.alphaz.application.authorization.service.UserService;
import com.alphaz.infrastructure.domain.model.common.ResponseModel;
import org.springframework.web.bind.annotation.*;

import javax.annotation.Resource;
import javax.validation.Valid;

/**
 * Created by C0dEr on 2016/12/3.
 */
@RequestMapping("/user")
@RestController
public class UserController {
    @Resource
    UserService userService;
    @Resource
    PrivilegeService privilegeService;


    @GetMapping("/{username}/role")
    public ResponseModel getRoleByUser(@PathVariable String username) {
        return this.privilegeService.getUserRoleByUsername(username);
    }

    @GetMapping("/{id}/menu/op")
    public ResponseModel getMOByUser(@PathVariable Long id) {
        return this.privilegeService.getMOByUserid(id);
    }

    @PostMapping("/new")
    public ResponseModel addUser(@Valid UserModel model) {
        return userService.addUser(model);
    }

    @PostMapping("/forgetpassword")
    public ResponseModel forgetPassword(String username, String password) {
        return userService.changePassword(username, password);
    }

    @PatchMapping("/{userid}")
    public ResponseModel updateUser(@PathVariable Long userid, @Valid UserUpdateModel model) {
        return userService.updateUser(userid, model);
    }

    @DeleteMapping("/{userid}")
    public ResponseModel deleteUser(@PathVariable Long userid) {
        return this.userService.deleteUser(userid);
    }

//    @GetMapping("/search")
//    public DataTableModel findUser(@Valid UserSearchModel model) {
//        return this.userService.findUser(model);
//    }
}

File: Api/src/main/kotlin/com/alphaz/api/controller/UserController.kt
package com.alphaz.api.controller

import com.alphaz.application.authorization.user.UserAppService
import com.alphaz.application.authorization.user.dto.UserDto
import com.alphaz.core.authorization.user.User
import com.alphaz.infrastructure.web.base.BaseControllerImpl
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/**
 *@Author: c0der
 *@Date: 下午10:14 2018/4/1
 *@Description:
 */
@RestController
@RequestMapping("user")
@PreAuthorize("permitAll()")
open class UserController : BaseControllerImpl<UserDto, User, Long, UserAppService>() {
    @PostMapping("kkk")
    open fun a(uuu: Boolean): Boolean {
        return false;
    }
}